protocols/nsclient/src/main/java/org/opennms/protocols/nsclient/config/NSClientPeerFactory.java

```diff
@@ -131,6 +139,29 @@ public Lock getWriteLock() {
         return m_writeLock;
     }
 
+    private synchronized Scope getSecureCredentialsScope() {
```

Comment:
Don't need to do this in this PR, but if we are using this code in multiple places, perhaps make a ticket to create a component or factory that does it and include just that.

Reply:
What do you mean? The method is called in this class?

Reply:
It looks like the same code as in `WmiPeerFactory`, my comment was just that if this code is in 3 or more places it should be factored out if possible. No need to do that in this PR, but it could be something to do separately.

Reply:
Ah. Got it.